x-pack/legacy/plugins/integrations_manager/server/config.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+
+export const config = {
+  schema: schema.object({
+    enabled: schema.boolean(),
+    registryUrl: schema.string(),
+  }),
+};
+
+export type IntegrationsManagerConfigSchema = TypeOf<typeof config.schema>;
+
+const DEFAULT_CONFIG = {
+  enabled: true,
+  registryUrl: 'http://integrations-registry.app.elstc.co',
+};
+
+// As of 2019, this is a Singleton because of the way JavaScript modules are specified.
+// Every other module that imports this file will have access to the same object.
+
+// This is meant to be only updated from the config$ Observable's subscription
+// (see the Plugin class constructor in server/plugin.ts) but this is not enforced.
+
+class IntegrationsManagerConfigStore {
+  private config: IntegrationsManagerConfigSchema;
+
+  constructor() {
+    this.config = DEFAULT_CONFIG;
+  }
+
+  updateConfig(newConfig: IntegrationsManagerConfigSchema) {
```

Comment:
Is `newConfig` the entire config object or only the updated keys/values?

Reply:
Good question. Right now this is a shim, so I don't know how the real NP implementation works or will work -- @rudolf, can you comment on this?

Reply:
Looking at this again, I'm assuming it's the full `IntegrationsManagerConfigSchema` object with updated values since it subscribes to the `config$` Observable.

Reply:
That's my assumption as well.